go-controller/pkg/ovn/address_set.go

```diff
@@ -17,12 +17,27 @@ type AddressSetDoFunc func(as AddressSet) error
 
 // AddressSetFactory is an interface for managing address set objects
 type AddressSetFactory interface {
-	// NewAddressSet returns a new object that implements NewAddressSet
+	// NewAddressSet returns a new object that implements AddressSet
 	// and contains the given IPs, or an error
-	NewAddressSet(name string, ips []net.IP) (AddressSet, error)
+	NewAddressSet(name string, ips []*net.IP) (AddressSet, error)
```

Comment:
IMO using `*net.IP` is confusing, and if we don't have actual data suggesting that saving a few bytes makes a difference, I would rather go with the idiomatic version.

Reply:
@abhat requested this change, so I'll let him respond. I don't really care either way.

Reply:
I'm happy to change back, after discussing with Tim on Friday the only case where this would be a performance issue is likely at namespace creation time when grabbing pod IPs and copying their array. But that array will only be large on restart anyway.

Reply:
Changed back

Reply:
> IMO using `*net.IP` is confusing, and if we don't have actual data suggesting that saving a few bytes makes a difference, I would rather go with the idiomatic version.

Why is that confusing? I can understand the readability problem. But unless there is a pressing reason beyond readability (like we truly don't want to pass something that is held internally in a cache and want to pass copies around), I see value in being frugal. Having said that, I am ok with this for now given @dcbw has already changed this back and forth a couple of times.

Reply:
> Why is that confusing?

Because it just looks wrong. You don't normally have pointers to pointer types. And so people looking at the code will assume there must be some semantic reason for it, when in fact there isn't.